### DISCUSSION TITLE
Application Access Session Recording

### DISCUSSION THREAD

**Discussion by jiahao19971** [gravitational/teleport#12817] [Q&A] [answered]
Hi, would like to ask if we used application access to kubernetes-dashboard and we exec into a pod, it seems like it doesn't get recorded into session recordings but instead it return a message. Would like to ask if this is a config error or application access cannot be used to do session recording ?
```
ERROR REPORT:
Original Error: *trace.NotFoundError open /var/lib/teleport/log/upload/streaming/default/multi/b3dda06d-3452-4f2b-b9e6-4b9944c4e664/8497d1d7-2719-4400-8eed-903f6e674bee/1.part: no such file or directory
Stack Trace:
	/go/src/github.com/gravitational/teleport/lib/events/filesessions/filestream.go:83 github.com/gravitational/teleport/lib/events/filesessions.(*Handler).UploadPart
	/go/src/github.com/gravitational/teleport/lib/events/stream.go:664 github.com/gravitational/teleport/lib/events.(*sliceWriter).startUpload.func1
	/opt/go/src/runtime/asm_amd64.s:1581 runtime.goexit
User Message: open /var/lib/teleport/log/upload/streaming/default/multi/b3dda06d-3452-4f2b-b9e6-4b9944c4e664/8497d1d7-2719-4400-8eed-903f6e674bee/1.part: no such file or directory] events/stream.go:493
```

**Comment by scottpgallagher** [comment, marked as answer]
It should work; you can follow the instructions on this github response regarding how to use it: 
https://github.com/gravitational/teleport/discussions/11334

See if that works.

**Reply by jiahao19971** [reply]
I couldn't find the records folder, all I can see after I exec into the pod `cd var/lib/teleport/log` are just playbacks and upload. Within these folder, there are file called 1.part which I have no idea how to access it.

**Reply by webvictim** [reply]
The session recordings are located on your Teleport auth/proxy server, not the pod which runs the agent. I would double check that you're connecting to the correct location.

**Reply by jiahao19971** [reply]
Yeap, accessing to the auth/proxy server not the agent, but couldn't find the directory, instead i change the `session_recording: proxy-sync` and it seems like it is able to upload to the s3 buckets and the issue doesn't pop out anymore

**Reply by jiahao19971** [reply]
So far the only way to read the application access session logging is via `tsh play --format ./<session_chunk_id>.tar` ? There isn't any clip type recording yet ?

**Reply by webvictim** [reply]
That is correct - at present there's no other way to play these recordings back. There will likely never be a video-type recording because it isn't possible to piece that together based on the logs Teleport has - it only sees requests, not a full browser view.
